### PR TITLE
Add support for fields substitutions.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: src/linker/Linker.Steps/BodySubstituterStep.cs [hunk 1]
```diff
@@ -157,7 +157,7 @@ void ProcessField (TypeDefinition type, XPathNodeIterator iterator)
 
 			var field = type.Fields.FirstOrDefault (f => f.Name == name);
 			if (field == null) {
-				Context.LogMessage (MessageImportance.High, $"Could not find field '{name}' for substitution.");
+				Context.LogMessage (MessageImportance.Low, $"Could not find field '{name}' for substitution.");
```
Comment (vargaz) [inline src/linker/Linker.Steps/BodySubstituterStep.cs]: Shouldn't all these cases be errors ?

Comment (marek-safar) [reply src/linker/Linker.Steps/BodySubstituterStep.cs]: I don't think we have to be pedantic if someone wants to use single xml for multiple purposes

Comment (vargaz) [reply src/linker/Linker.Steps/BodySubstituterStep.cs]: The problem is that these can silently stop working etc.

Comment (marek-safar) [reply src/linker/Linker.Steps/BodySubstituterStep.cs]: Yes but it's not a blocker. Let's go with .Normal

**Pull request merged**